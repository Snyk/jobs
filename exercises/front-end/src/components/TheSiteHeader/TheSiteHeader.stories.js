import '~/assets/scss/styles.scss';
import TheSiteHeader from './TheSiteHeader';

export default {
  title: 'Library/TheSiteHeader',
  component: TheSiteHeader,
};

export const Default = () => ({
  components: { TheSiteHeader },
  template: `<TheSiteHeader />`,
});
