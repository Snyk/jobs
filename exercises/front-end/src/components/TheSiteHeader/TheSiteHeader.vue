<template>
  <header class="vue--site-header">
    <LayoutContainer :class="computedClasses">
      <a href="/" class="vue--site-header__home-link">
        <SnykLogo
          class="vue--site-header__logo"
          fill-color="#fff"
          :size="52"
          data-snyk-test="TheSiteHeader: logo"
        />
      </a>
      <span class="vue--site-header__title">
        {{ title }}
      </span>
    </LayoutContainer>
  </header>
</template>

<script>
import LayoutContainer from '~/components/LayoutContainer/LayoutContainer';
import SnykLogo from '~/components/CustomIcons/Snyk';

export default {
  name: 'TheSiteHeader',
  components: {
    LayoutContainer,
    SnykLogo,
  },
  props: {
    title: {
      type: String,
    },
  },
  computed: {
    computedClasses() {
      return {
        'vue--site-header__container': true,
      };
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~/assets/scss/utils';

.vue--site-header {
  height: 64px;
  margin-bottom: space(l);

  background: linear-gradient(
    to right,
    color(gradient, purple-start) 0%,
    color(gradient, purple-end) 50%
  );

  &__container {
    overflow: visible;
    padding-top: space(xs);
    padding-bottom: space(s);

    position: relative;
    z-index: 100;
  }

  &__home-link {
    display: inline-block;
    padding-right: space(s);
    flex-grow: 2;

    &:focus-visible {
      outline: 2px solid color(ui, white);
      opacity: 1;
    }
  }

  &__title {
    color: color(ui, white);
    font-size: rem(22px);
    margin: 0.25em 0;

    @include media-query(s) {
      margin: 0.25em 0.5em;
    }
  }

  &__logo {
    position: relative;
    top: -3px;
  }
}
</style>
