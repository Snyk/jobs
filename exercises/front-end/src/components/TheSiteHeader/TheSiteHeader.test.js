import { shallowMount, mount } from '@vue/test-utils';
import TheSiteHeader from './TheSiteHeader';

describe('TheSiteHeader', () => {
  describe('Accessibility', () => {
    it('passes basic accessibility checks', async () => {
      const wrapper = shallowMount(TheSiteHeader);
      await expect(wrapper).toBeAccessible();
    });
  });

  describe('Snapshots', () => {
    it('renders as expected', () => {
      const wrapper = shallowMount(TheSiteHeader);
      expect(wrapper).toMatchSnapshot();
    });

    it('renders valid html', () => {
      const wrapper = mount(TheSiteHeader);
      expect(wrapper.html()).toHTMLValidate();
    });
  });
});
