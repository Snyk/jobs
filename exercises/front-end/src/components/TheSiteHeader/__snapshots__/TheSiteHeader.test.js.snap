// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheSiteHeader Snapshots renders as expected 1`] = `
<header class="vue--site-header">
  <layoutcontainer-stub class="vue--site-header__container"><a href="/" class="vue--site-header__home-link">
      <snyklogo-stub title="Snyk" fillcolor="#fff" size="52" data-snyk-test="TheSiteHeader: logo" class="vue--site-header__logo"></snyklogo-stub>
    </a> <span class="vue--site-header__title">

    </span></layoutcontainer-stub>
</header>
`;
